### PR TITLE
Introduce water leakage and tilt service

### PR DESCRIPTION
**Pull request by Ebolon**

### CODE REVIEW

**Comment by tschamm** [comment]
Thanks for introducing the water leakage sensor. Do you mind to add the belonging device model (c.f. https://github.com/tschamm/boschshcpy/blob/master/boschshcpy/models_impl.py and https://github.com/tschamm/boschshcpy/blob/master/boschshcpy/device_helper.py)?

**Comment by Ebolon** [comment]
It is not 100% complete because the configuration of the sensor from the app is missing. I could not figure out how to capture those events. Also I could not figure out the potential states of the sensor check service (maybe 'OK',...). So far I would rate the implementation as experimental.

File: boschshcpy/services_impl.py [hunk 1]
```diff
@@ -622,6 +622,20 @@ def summary(self):
         print(f"    acousticSignalState      : {self.acousticSignalState}")
 
 
+class WaterLeakageSensorCheckService(SHCDeviceService):
+    class State(Enum):
+        NONE = "NONE"
+
+    @property
+    def value(self) -> State:
+        return self.State(self.state["result"])
```
Comment (tschamm) [inline boschshcpy/services_impl.py:631]: Parsing the state of the sensor will result in an API exception when the self.state["result"] is not "NONE". If only the state string is used, error handling can be done when evaluating the actual state.

File: boschshcpy/models_impl.py [hunk 2]
```diff
@@ -660,6 +663,41 @@ def summary(self):
...
+        return self._sensor_check_service.value
+
+    def update(self):
+        self._leakage_service.short_poll()
+        self._tilt_service.short_poll()
+        self._sensor_check_service.short_poll()
+        super().update()
+
+    def summary(self):
+        print(f"SD WaterLeakageSensor:")
```
Comment (tschamm) [inline boschshcpy/models_impl.py:697]: Should be `print(f"WATER_LEAKAGE_SENSOR:")`

**Review by tschamm — COMMENTED** [review]
Thanks for adding the model device. If you can identify the other states for the sensor check service or just provide the state as string, I think the PR is ready for merging.

**Comment by tschamm** [comment]
> It is not 100% complete because the configuration of the sensor from the app is missing. I could not figure out how to capture those events. 

What kind of configuration are you missing? Normally, the configuration is stored in the device properties. Using postman, the device properties can be read with `https://{{host}}:8444/smarthome/devices/{{device_id}}`.

File: boschshcpy/services_impl.py (hunk 1 above)
Comment (Ebolon) [reply boschshcpy/services_impl.py]: Could not find any description for `/devices/<device_id>/services/WaterLeakageSensorCheck/state`. I changed the type to string until I find a way to fetch all available enum values.

File: boschshcpy/models_impl.py (hunk 2 above)
Comment (Ebolon) [reply boschshcpy/models_impl.py]: 👍

**Comment by Ebolon** [comment]
> > It is not 100% complete because the configuration of the sensor from the app is missing. I could not figure out how to capture those events.
> 
> What kind of configuration are you missing? Normally, the configuration is stored in the device properties. Using postman, the device properties can be read with `https://{{host}}:8444/smarthome/devices/{{device_id}}`.

I thought about the possible operations to switch the state e.g. is possible to turn on/off notifications. But for now I would keep the pull request as it.

**Comment by tschamm** [comment]
I'll increase the version number, so the new sensor can be accessed in home assistant as well.
Thanks for your PR!

File: boschshcpy/services_impl.py (hunk 1 above)
Comment (tschamm) [reply boschshcpy/services_impl.py]: You can either file an issue in the official Bosch API for updating the description, or just watch on state changes when putting the sensor into different modes (e.g. put water on the contacts and just watch how the device behaves in postman).

**Pull request merged**